### ISSUE TITLE
promise when don't work with php i think

### ISSUE THREAD

**Issue by francescoagati** [jdonaldson/promhx#71]
I think that when macro don't work with php. I think that the problem is for sync mode of php

**Comment by jdonaldson** [comment]
Yep, there's some issues there for php.  I currently have it in the "penalty box" for allowed failures:
https://travis-ci.org/jdonaldson/promhx

I've been meaning to bother Franco about this, but I don't use the php target very much.  Any ideas on what might be going on there would be much appreciated.

**Comment by jdonaldson** [comment]
Side note:  php shows as "passing" in travis, but in fact it is not... I need to figure out what the issue is there as well.

**Comment by francescoagati** [comment]
i will try to know the problem and modify the macro with a pull request. The problem is that whenAll works good in php, but the macro when don't work. The macro when don't call the callback when all the promises are resolved while when trigger the callback
